jobserver/api.py

```diff
@@ -273,3 +276,45 @@ def get(self, request, *args, **kwargs):
 
         actions_with_status = workspace.get_action_status_lut()
         return Response(actions_with_status, status=200)
+
+
+class ReleaseUploadAPI(APIView):
+    # DRF file upload does not use multipart, is just a simple byte stream
+    parser_classes = [FileUploadParser]
+
+    def post(self, request, workspace_name):
+        try:
+            workspace = Workspace.objects.get(name=workspace_name)
+        except Workspace.DoesNotExist:
+            return Response(status=404)
+
+        # authenticate and get backend
+        backend = get_backend_from_token(request.headers.get("Authorization"))
+        backend_user = request.headers.get("Backend-User", "").strip()
+        if not backend_user:
+            raise NotAuthenticated("Backend-User not valid")
+
+        # TODO: validate user once we have mappings from backend username.
+
+        release_hash = request.headers.get("If-None-Match")
+        if release_hash is None:
+            raise ValidationError("No If-None-Match header with release hash")
+
+        if "file" not in request.data:
+            raise ValidationError("No data uploaded")
+
+        upload = request.data["file"]
+        release, created = handle_release(
+            workspace, backend, backend_user, release_hash, upload
+        )
+
+        response = Response(status=201 if created else 303)
+        response["Location"] = urls.reverse(
```

Comment:
`urls.reverse` is typically imported as `reverse` directly, could you do that here for consistency?

Reply:
done